src/Graphs/partitionedgraphs/abstractpartitionedgraph.jl

```diff
@@ -0,0 +1,173 @@
+  add_edge!(unpartitioned_graph(pg), edge)
+  pg_edge = parent(partition_edge(pg, edge))
+  if src(pg_edge) != dst(pg_edge)
+    add_edge!(partitioned_graph(pg), pg_edge)
+  end
+end
+
+function rem_edge!(pg::AbstractPartitionedGraph, edge::AbstractEdge)
+  pg_edge = partition_edge(pg, edge)
+  if has_edge(partitioned_graph(pg), pg_edge)
+    g_edges = edges(pg, pg_edge)
+    if length(g_edges) == 1
+      rem_edge!(partitioned_graph(pg), pg_edge)
+    end
+  end
+
+  return rem_edge!(unpartitioned_graph(pg), edge)
+end
+
+function rem_edge!(pg::AbstractPartitionedGraph, partition_edge::AbstractPartitionEdge)
+  return rem_edges!(pg, edges(pg, parent(partition_edge)))
+end
+
+function rem_edge(pg::AbstractPartitionedGraph, partition_edge::AbstractPartitionEdge)
+  pg_new = copy(pg)
+  rem_edge!(pg_new, partition_edge)
+  return pg_new
+end
+
+function rem_edges!(
+  pg::AbstractPartitionedGraph, partition_edges::Vector{<:AbstractPartitionEdge}
+)
+  for pe in partition_edges
+    rem_edge!(pg, pe)
+  end
+end
+
+function rem_edges(
+  pg::AbstractPartitionedGraph, partition_edges::Vector{<:AbstractPartitionEdge}
+)
+  pg_new = copy(pg)
+  rem_edges!(pg_new, partition_edges)
+  return pg_new
+end
+
+#Vertex addition and removal. I think it's important not to allow addition of a vertex without specification of PV
+function add_vertex!(
+  pg::AbstractPartitionedGraph, vertex, partition_vertex::AbstractPartitionVertex
+)
+  add_vertex!(unpartitioned_graph(pg), vertex)
+  add_vertex!(partitioned_graph(pg), parent(partition_vertex))
+
+  return insert_to_vertex_map!(pg, vertex, partition_vertex)
+end
+
+function add_vertices!(
+  pg::AbstractPartitionedGraph,
+  vertices::Vector,
+  partition_vertices::Vector{<:AbstractPartitionVertex},
+)
+  @assert length(vertices) == length(partition_vertices)
+  for (v, pv) in zip(vertices, partition_vertices)
+    add_vertex!(pg, v, pv)
+  end
+end
+
+function add_vertices!(
+  pg::AbstractPartitionedGraph, vertices::Vector, partition_vertex::AbstractPartitionVertex
+)
+  return add_vertices!(pg, vertices, fill(partition_vertex, length(vertices)))
+end
+
+function rem_vertex!(pg::AbstractPartitionedGraph, vertex)
+  rem_vertex!(unpartitioned_graph(pg), vertex)
+  pv = which_partition(pg, vertex)
+  if length(vertices(pg, pv)) == 1
+    rem_vertex!(partitioned_graph(pg), parent(pv))
+  end
+  return delete_from_vertex_map!(pg, vertex)
+end
+
+function rem_vertex!(
+  pg::AbstractPartitionedGraph, partition_vertex::AbstractPartitionVertex
+)
+  return rem_vertices!(pg, vertices(pg, partition_vertex))
+end
+
+function rem_vertex(pg::AbstractPartitionedGraph, partition_vertex::AbstractPartitionVertex)
+  pg_new = copy(pg)
+  rem_vertex!(pg_new, partition_vertex)
+  return pg_new
+end
+
+function rem_vertices!(
+  pg::AbstractPartitionedGraph, partition_vertices::Vector{<:AbstractPartitionVertex}
+)
+  for pv in partition_vertices
+    rem_vertex!(pg, pv)
+  end
```

Comment:
```suggestion
  end
  return pg
```
Usually I return the object being modified as a convention.